packages/stylelint-config/src/index.js

```diff
@@ -13,7 +13,10 @@ module.exports = {
     'stylelint-config-css-modules',
 
     // Enable prettier formatting for SCSS/CSS
-    'stylelint-prettier/recommended',
+    // TODO: Not sure why this is causing rules to fail in tests. But they are.
+    // Do we need this if we use prettierrc files anyway separate from
+    // stylelint? Consider removing this extension if possible.
+    // 'stylelint-prettier/recommended',
```

Comment:
Still unsure about this one. This was messing up all the tests but not sure why.

Since we already have prettier setup though separately from stylelint, I'm tempted to just remove this extension

Reply:
@dannobytes I believe these tests are all failing with this extension being enabled because it seems like these tests all have valid issues.

I'm happy to dump this extension and just rely on Prettier looking at `.scss` files but fwiw not having this extension here won't surface those issues until after the file has been saved and you run `npm run prettier`. I don't think there's any way to integrate the Prettier VSCode extension with `stylelint.vscode-stylelint`.

// @kanadgupta

Reply:
@dannobytes what happens if you change [this file](https://github.com/readmeio/standards/blob/493a0d71969ab2df509f1fe6c23dfeea39deb7cb/.prettierrc.cjs) to be `.mjs`? Is it possible that it's not getting picked up by stylelint?

Reply:
@kanadgupta @dannobytes I just pushed https://github.com/readmeio/standards/pull/717/commits/fc10aeeaeabcf1c041089fb6ba8a7dbac4653ed4 which fixes these broken tests and re-enables `stylelint-prettier`. The `valid.scss` file wasn't actually valid anymore with the Prettier bump and that was causing all these failures.

Reply:
Sorry @erunion I should have been more clear. But the snapshots you updated are actually all wrong and is auto-fixing them as if none of our stylelint rules are in effect. I added some inline comments to help point them out.

For some reason, when re-enabling the `stylelint-prettier` extension, it seems to nullify or ignore all our inline stylelint rules. I'm wondering if it's just silently error'ing out and causing the rest of the config rules to not get parsed. But either way, this is the main problem.

Our existing snapshots are actually correct and we need to preserve what they are.

Reply:
> @dannobytes what happens if you change [this file](https://github.com/readmeio/standards/blob/493a0d71969ab2df509f1fe6c23dfeea39deb7cb/.prettierrc.cjs) to be `.mjs`? Is it possible that it's not getting picked up by stylelint?

@kanadgupta I only added that top-level prettierrc file b/c I was noticing in my own editor, that it didn't quite know how to format files under `packages/stylelint` when saving. After adding it, it all seemed to be fine for me locally and didn't impact any of the test runs at all.

But I am a little confused by this all ESM switch that prettier did in v3. I'm sure this has something to do with how/whether the extension is getting parsed correctly by `stylelint` or not.

Reply:
> Our existing snapshots are actually correct and we need to preserve what they are.

@dannobytes isn't the intent of `stylelint-prettier` to adhere to our existing Prettier configuration? If so then some of the changes in https://github.com/readmeio/standards/pull/717/commits/fc10aeeaeabcf1c041089fb6ba8a7dbac4653ed4 were needed, right? For example our Prettier config is set to prefer single quotes:

https://github.com/readmeio/standards/blob/fc10aeeaeabcf1c041089fb6ba8a7dbac4653ed4/packages/eslint-config/prettier.js#L4

and that's what fc10aeeaeabcf1c041089fb6ba8a7dbac4653ed4 changes the snapshot to use:

<img width="549" alt="CleanShot 2023-08-02 at 18 08 02@2x" src="https://github.com/readmeio/standards/assets/8854718/21362940-23ac-49a8-8684-4784ffe627dd">

I do see [a section in the `stylelint-prettier` docs about disabling rules that may conflict with Prettier](https://github.com/prettier/stylelint-prettier#disabling-rules-that-may-conflict-with-prettier), could that be the issue?

I'm pretty sure we had a similar issue with ESLint and Prettier clashing and that's why we introduced [`eslint-config-prettier`](https://github.com/prettier/eslint-config-prettier) into the mix.

Reply:
Yep, that snapshot update, where it's just double-quotes to single-quotes. But the other css related auto-fixes are not. So I should have been more clear, that not *all* the snapshot updates are incorrect. Just the CSS ones that seem to be ignoring our ruleset whenever stylelint/prettier is enabled.

Reply:
I'll revert the snapshot updates I put in but to be honest, looking through the `stylelint-prettier` source I don't see where it actually does any ignores based on the `stylelintrc`. https://github.com/prettier/stylelint-prettier/blob/main/stylelint-prettier.js

Seems like it's always just running Prettier and reporting back whatever the problem is, period.

I'm also a little concerned that if we disable this plugin and then run Prettier on our `.scss` files that it'll do the same that the plugin is doing now.

Reply:
Yeah, I agree we should definitely not disable or remove the prettier plugin at this point. But I'm definitely confused at the moment as to what's causing stylelint rules to change whenever the "prettier/prettier" rule is turned on.

Was going to dig into that a bit more because what I noticed today when troubleshooting was that the problem has less to do with adding the prettier extension but more when the prettier rule is enabled. Like if I set `prettier/prettier: false` under the rules section, stylelint runs as it should.

Will look into this more